kong/plugins/aws-lambda/handler.lua

```diff
@@ -19,20 +31,62 @@ function AWSLambdaHandler:new()
   AWSLambdaHandler.super.new(self, "aws-lambda")
 end
 
-local function retrieve_parameters()
-  ngx_req_read_body()
-
-  return utils.table_merge(ngx_req_get_uri_args(), public_utils.get_body_args())
-end
-
 function AWSLambdaHandler:access(conf)
   AWSLambdaHandler.super.access(self)
 
-  local bodyJson = cjson.encode(retrieve_parameters())
+  local upstream_body = new_tab(0, 6)
+
+  do
+    ngx_req_read_body()
+
+    local body_args, err_code, body_raw = public_utils.get_body_info()
+    if err_code == public_utils.req_body_errors.unknown_ct then
+      -- don't know what this body MIME type is, base64 it just in case
+      body_raw = ngx_encode_base64(body_raw)
```

Comment:
@thibaultcha "upstream_body.request_body.base64 = true" - sounds good. The use case for sending binary to lambda is replacing with lambda some existing services that are using binary protocols (or just compressed data to decrease traffic) for instance: fastinfoset, avro, protobuf. Also AWS API Gateway supports binary content https://aws.amazon.com/blogs/compute/binary-support-for-api-integrations-with-amazon-api-gateway/